plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/dlf/DlfMetaStoreClient.java

```diff
@@ -0,0 +1,1538 @@
+import static org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW;
+
+/**
+ * Data Lake Formation MetaStore Client for Trino.
+ */
+public class DlfMetaStoreClient
+        implements HiveMetastore
+{
+    private static final Logger logger = Logger.get(DlfMetaStoreClient.class);
+    private static final String PUBLIC_ROLE_NAME = "public";
+    private static final String DEFAULT_METASTORE_USER = "presto";
+    private static final String EMPTY_TOKEN = "";
+    private static final int NO_BATCHING = -1;
+    private final String catalogId;
+    //private DataLakeClient dataLakeClient;
+    private final DlfMetaStoreConfig dlfConfig;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HdfsContext hdfsContext;
+    private final Optional<String> defaultDir;
+    private final IDataLakeMetaStore dataLakeMetaStore;
+    private short pageSize;
+    private int batchSizeForGetPartititon;
+    private boolean isProxyModeDlfOnly;
+    private boolean enableFsOperation;
+
+    @Inject
+    public DlfMetaStoreClient(
+            HdfsEnvironment hdfsEnvironment,
+            DlfMetaStoreConfig dlfConfig) throws TrinoException
+    {
+        logger.info("DlfMetaStoreClient start");
+        requireNonNull(dlfConfig, "dlfConfig is null");
+
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.hdfsContext = new HdfsContext(ConnectorIdentity.ofUser(DEFAULT_METASTORE_USER));
+
+        this.dlfConfig = dlfConfig;
+        this.catalogId = DataLakeConfig.DEFAULT_CATALOG_ID;
+        this.pageSize = dlfConfig.getPageSize();
+        this.defaultDir = dlfConfig.getDefaultWarehouseDir();
+        this.dataLakeMetaStore = new MetastoreFactory().getMetaStore(dlfConfig);
+        this.batchSizeForGetPartititon = dlfConfig.getBatchSizeForGetPartititon().orElse(DataLakeConfig.DEFAULT_CATALOG_COL_STATS_PAGE_SIZE);
+
+        this.isProxyModeDlfOnly = dlfConfig.getProxyMode().orElse(ProxyMode.DLF_ONLY) == ProxyMode.DLF_ONLY;
+        this.enableFsOperation = this.isProxyModeDlfOnly || (dlfConfig.getEnableFileOperation().orElse(false) && DataLakeUtil.isEnableFileOperationGray(dlfConfig.getEnableFileOperationGrayRate().orElse(0.0d)));
+        logger.info("DlfMetaStoreClient end");
+        //reInitDataLakeClient(dlfConfig);
+    }
+
+    private static List<String> buildPartitionNames(List<Column> partitionColumns, List<Partition> partitions)
+    {
+        return partitions.stream()
+                .map(partition -> makePartitionName(partitionColumns, partition.getValues()))
+                .collect(Collectors.toList());
+    }
+
+    public IDataLakeMetaStore getDataLakeMetaStore()
+    {
+        return dataLakeMetaStore;
+    }
+
+    @Override
+    public Optional<Database> getDatabase(String databaseName)
+    {
+        return getDatabase(catalogId, databaseName);
+    }
+
+    public Optional<Database> getDatabase(String catalogId, String databaseName)
+    {
+        try {
```

Comment:
Can we collapse these into a single try/catch block, rather than nesting them?

Reply:
Come up with a way to solve it, PTAL.